From: Jeff Mahoney <jeffm@suse.com>
Subject: lookup_create should pass nameidata to lookup_hash

My reworked subfs patch was failing due to lookup_create passing a NULL
nameidata, even though it has a valid nameidata. This was causing subfs
to Oops when it tried to get nd->mnt during mkdir/mknod/link/symlink.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>

Index: linux-2.6.11/fs/namei.c
===================================================================
--- linux-2.6.11.orig/fs/namei.c
+++ linux-2.6.11/fs/namei.c
@@ -1524,7 +1524,7 @@ struct dentry *lookup_create(struct name
 	if (nd->last_type != LAST_NORM)
 		goto fail;
 	nd->flags &= ~LOOKUP_PARENT;
-	dentry = lookup_hash(&nd->last, nd->dentry);
+	dentry = __lookup_hash(&nd->last, nd->dentry, nd);
 	if (IS_ERR(dentry))
 		goto fail;
 	if (!is_dir && nd->last.name[nd->last.len] && !dentry->d_inode)
